### ISSUE TITLE
Public IP not detected.  If private IP is meant for bootstrap you must specify "ssh_interface: private_ips" in your profile.

### ISSUE THREAD

**Issue by lewayotte** [saltstack/salt#34627]
I'm trying to launch EC2 instances with salt-cloud, I want the SaltStack to be managed using the private IP addresses, but I also need the instances to have public Elastic IP addresses. This seems to work fine, except that I get this error message:

```
salt-cloud -p one_site six.domain.com
[ERROR   ] Public IP not detected.  If private IP is meant for bootstrap you must specify "ssh_interface: private_ips" in your profile.
```

I do have `ssh_interface: private_ips` specified in the profile and the provider.

cloud.profiles.d/domain.conf:
```
base:
  provider: domain
  network_interfaces:
    - DeviceIndex: 0
      PrivateIpAddresses:
        - Primary: True
      SubnetId: subnet-XXXXXXXX
      SecurityGroupId:
        - sg-XXXXXXXX
      allocate_new_eips: True
  ssh_interface: private_ips
  ssh_username: ubuntu
  del_root_vol_on_destroy: True
  del_all_vol_on_destroy: True

one_site:
  extends: base
  image: ami-XXXXXXXX
  size: t2.micro

three_sites:
  extends: base
  image: ami-XXXXXXXX
  size: t2.small

ten_sites:
  extends: base
  image: ami-XXXXXXXX
  size: t2.medium
```
cloud.providers.d/aws.conf 
```
domain:
  id: 'MY_ID'
  key: 'MY_KEY'
  keyname: 'KEY_NAME'
  private_key: /root/.ssh/KEY_FILE.pem
  driver: ec2
  minion: 
    master: XXX.XXX.XXX.XXX
  ssh_interface: private_ips
  ssh_userame: ubuntu
```

Versions Report:
```
Salt Version:
           Salt: 2016.3.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-74-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

Debug:
salt-cloud -p one_site six.domain.com -l debug
```
[DEBUG   ] Reading configuration from /etc/salt/cloud
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: ip-172-31-59-214.ec2.internal
[DEBUG   ] Missing configuration file: /etc/salt/cloud.providers
[DEBUG   ] Including configuration from '/etc/salt/cloud.providers.d/aws.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.providers.d/aws.conf
[DEBUG   ] Missing configuration file: /etc/salt/cloud.profiles
[DEBUG   ] Including configuration from '/etc/salt/cloud.profiles.d/domain.conf'
[DEBUG   ] Reading configuration from /etc/salt/cloud.profiles.d/domain.conf
[DEBUG   ] Configuration file path: /etc/salt/cloud
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] salt-cloud starting
[DEBUG   ] Could not LazyLoad parallels.avail_sizes
[DEBUG   ] LazyLoaded parallels.avail_locations
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.destroy
[DEBUG   ] Could not LazyLoad saltify.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.avail_images
[DEBUG   ] Could not LazyLoad saltify.avail_locations
[DEBUG   ] LazyLoaded rackspace.reboot
[DEBUG   ] LazyLoaded openstack.list_locations
[DEBUG   ] LazyLoaded rackspace.list_locations
[DEBUG   ] Could not LazyLoad parallels.avail_sizes
[DEBUG   ] LazyLoaded parallels.avail_locations
[DEBUG   ] LazyLoaded proxmox.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.destroy
[DEBUG   ] Could not LazyLoad saltify.avail_sizes
[DEBUG   ] Could not LazyLoad saltify.avail_images
[DEBUG   ] Could not LazyLoad saltify.avail_locations
[DEBUG   ] LazyLoaded rackspace.reboot
[DEBUG   ] LazyLoaded openstack.list_locations
[DEBUG   ] LazyLoaded rackspace.list_locations
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeInstances&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeInstances&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Generating minion keys for 'six.domain.com'
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'profile': 'one_site', 'event': 'starting create', '_stamp': '2016-07-13T02:31:58.922758', 'name': 'six.domain.com', 'provider': 'domain:ec2'}
[INFO    ] Creating Cloud VM six.domain.com in us-east-1
[DEBUG   ] Create network interface: {'SubnetId': 'subnet-XXXXXXXX', 'DeviceIndex': 0, 'SecurityGroupId': ['sg-XXXXXXXXX'], 'PrivateIpAddresses': [{'Primary': True}], 'allocate_new_eips': True}
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeSubnets&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeSubnets&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=CreateNetworkInterface&PrivateIpAddresses.0.Primary=true&SecurityGroupId.0=sg-XXXXXXXXX&SubnetId=subnet-XXXXXXXX&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=CreateNetworkInterface&PrivateIpAddresses.0.Primary=true&SecurityGroupId.0=sg-XXXXXXXXX&SubnetId=subnet-XXXXXXXX&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Created network interface eni-63c8e37d inst 0
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=AllocateAddress&Domain=vpc&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=AllocateAddress&Domain=vpc&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=AssociateAddress&AllocationId=eipalloc-a0f13e9d&NetworkInterfaceId=eni-63c8e37d&PrivateIpAddress=172.31.59.85&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=AssociateAddress&AllocationId=eipalloc-a0f13e9d&NetworkInterfaceId=eni-63c8e37d&PrivateIpAddress=172.31.59.85&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Associated ElasticIP address eipalloc-a0f13e9d with interface eni-63c8e37d
[INFO    ] Attempting to look up root device name for image id ami-XXXXXXXX on VM six.domain.com
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeImages&ImageId.1=ami-XXXXXXXX&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeImages&ImageId.1=ami-XXXXXXXX&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] EC2 Response: '[{'virtualizationType': 'hvm', 'blockDeviceMapping': {'item': {'deviceName': '/dev/sda1', 'ebs': {'snapshotId': 'snap-7cdffa9a', 'deleteOnTermination': 'false', 'volumeSize': '20', 'volumeType': 'gp2', 'encrypted': 'false'}}}, 'name': '1 Site', 'imageOwnerId': '735935705556', 'hypervisor': 'xen', 'isPublic': 'false', 'sriovNetSupport': 'simple', 'imageId': 'ami-XXXXXXXX', 'imageState': 'available', 'architecture': 'x86_64', 'tagSet': {'item': {'value': 'June 2016', 'key': 'Updated'}}, 'imageLocation': '735935705556/1 Site', 'rootDeviceType': 'ebs', 'creationDate': '2016-06-09T02:39:59.000Z', 'rootDeviceName': '/dev/sda1', 'imageType': 'machine', 'description': 'domain Single Site Instance'}]'
[INFO    ] Found root device name: /dev/sda1
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'event': 'requesting instance', '_stamp': '2016-07-13T02:31:59.698380', 'location': 'us-east-1', 'kwargs': {'NetworkInterface.0.DeviceIndex': 0, 'BlockDeviceMapping.0.Ebs.VolumeType': 'gp2', 'ImageId': 'ami-XXXXXXXX', 'KeyName': 'KEY_NAME', 'MaxCount': 1, 'Action': 'RunInstances', 'MinCount': 1, 'BlockDeviceMapping.0.DeviceName': '/dev/sda1', 'NetworkInterface.0.NetworkInterfaceId': 'eni-63c8e37d', 'InstanceType': 't2.micro', 'BlockDeviceMapping.0.Ebs.DeleteOnTermination': 'true'}}
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=RunInstances&BlockDeviceMapping.0.DeviceName=%2Fdev%2Fsda1&BlockDeviceMapping.0.Ebs.DeleteOnTermination=true&BlockDeviceMapping.0.Ebs.VolumeType=gp2&ImageId=ami-XXXXXXXX&InstanceType=t2.micro&KeyName=KEY_NAME&MaxCount=1&MinCount=1&NetworkInterface.0.DeviceIndex=0&NetworkInterface.0.NetworkInterfaceId=eni-63c8e37d&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=RunInstances&BlockDeviceMapping.0.DeviceName=%2Fdev%2Fsda1&BlockDeviceMapping.0.Ebs.DeleteOnTermination=true&BlockDeviceMapping.0.Ebs.VolumeType=gp2&ImageId=ami-XXXXXXXX&InstanceType=t2.micro&KeyName=KEY_NAME&MaxCount=1&MinCount=1&NetworkInterface.0.DeviceIndex=0&NetworkInterface.0.NetworkInterfaceId=eni-63c8e37d&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'instance_id': 'i-e3b2477c', '_stamp': '2016-07-13T02:32:00.505167', 'event': 'querying instance'}
[DEBUG   ] The new VM instance_id is i-e3b2477c
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeInstances&InstanceId.1=i-e3b2477c&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeInstances&InstanceId.1=i-e3b2477c&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] The query returned: [{'ownerId': '735935705556', 'groupSet': None, 'reservationId': 'r-b149330e', 'instancesSet': {'item': {'productCodes': None, 'vpcId': 'vpc-9bd87aff', 'instanceId': 'i-e3b2477c', 'imageId': 'ami-XXXXXXXX', 'keyName': 'KEY_NAME', 'clientToken': None, 'subnetId': 'subnet-XXXXXXXX', 'amiLaunchIndex': '0', 'instanceType': 't2.micro', 'groupSet': {'item': {'groupName': 'LEMP Stack', 'groupId': 'sg-XXXXXXXXX'}}, 'monitoring': {'state': 'disabled'}, 'dnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'privateIpAddress': '172.31.59.85', 'virtualizationType': 'hvm', 'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'reason': None, 'sourceDestCheck': 'true', 'blockDeviceMapping': None, 'placement': {'groupName': None, 'tenancy': 'default', 'availabilityZone': 'us-east-1c'}, 'instanceState': {'code': '0', 'name': 'pending'}, 'networkInterfaceSet': {'item': {'status': 'in-use', 'macAddress': '12:ce:24:6d:30:9b', 'sourceDestCheck': 'true', 'vpcId': 'vpc-9bd87aff', 'description': None, 'networkInterfaceId': 'eni-63c8e37d', 'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'groupSet': {'item': {'groupName': 'LEMP Stack', 'groupId': 'sg-XXXXXXXXX'}}, 'attachment': {'status': 'attaching', 'deviceIndex': '0', 'deleteOnTermination': 'false', 'attachmentId': 'eni-attach-209c3afd', 'attachTime': '2016-07-13T02:32:00.000Z'}, 'subnetId': 'subnet-XXXXXXXX', 'ownerId': '735935705556', 'privateIpAddressesSet': {'item': {'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'association': {'publicIp': '52.20.118.88', 'publicDnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'ipOwnerId': '735935705556'}, 'primary': 'true', 'privateIpAddress': '172.31.59.85'}}, 'association': {'publicIp': '52.20.118.88', 'publicDnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'ipOwnerId': '735935705556'}, 'privateIpAddress': '172.31.59.85'}}, 'ebsOptimized': 'false', 'launchTime': '2016-07-13T02:32:00.000Z', 'architecture': 'x86_64', 'hypervisor': 'xen', 'rootDeviceType': 'ebs', 'ipAddress': '52.20.118.88', 'rootDeviceName': '/dev/sda1'}}}]
[DEBUG   ] Waiting for VM IP. Giving up in 00:10:00.
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeInstances&InstanceId.1=i-e3b2477c&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeInstances&InstanceId.1=i-e3b2477c&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Returned query data: [{'ownerId': '735935705556', 'groupSet': None, 'reservationId': 'r-b149330e', 'instancesSet': {'item': {'productCodes': None, 'vpcId': 'vpc-9bd87aff', 'instanceId': 'i-e3b2477c', 'imageId': 'ami-XXXXXXXX', 'keyName': 'KEY_NAME', 'clientToken': None, 'subnetId': 'subnet-XXXXXXXX', 'amiLaunchIndex': '0', 'instanceType': 't2.micro', 'groupSet': {'item': {'groupName': 'LEMP Stack', 'groupId': 'sg-XXXXXXXXX'}}, 'monitoring': {'state': 'disabled'}, 'dnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'privateIpAddress': '172.31.59.85', 'virtualizationType': 'hvm', 'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'reason': None, 'sourceDestCheck': 'true', 'blockDeviceMapping': None, 'placement': {'groupName': None, 'tenancy': 'default', 'availabilityZone': 'us-east-1c'}, 'instanceState': {'code': '0', 'name': 'pending'}, 'networkInterfaceSet': {'item': {'status': 'in-use', 'macAddress': '12:ce:24:6d:30:9b', 'sourceDestCheck': 'true', 'vpcId': 'vpc-9bd87aff', 'description': None, 'networkInterfaceId': 'eni-63c8e37d', 'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'groupSet': {'item': {'groupName': 'LEMP Stack', 'groupId': 'sg-XXXXXXXXX'}}, 'attachment': {'status': 'attaching', 'deviceIndex': '0', 'deleteOnTermination': 'false', 'attachmentId': 'eni-attach-209c3afd', 'attachTime': '2016-07-13T02:32:00.000Z'}, 'subnetId': 'subnet-XXXXXXXX', 'ownerId': '735935705556', 'privateIpAddressesSet': {'item': {'privateDnsName': 'ip-172-31-59-85.ec2.internal', 'association': {'publicIp': '52.20.118.88', 'publicDnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'ipOwnerId': '735935705556'}, 'primary': 'true', 'privateIpAddress': '172.31.59.85'}}, 'association': {'publicIp': '52.20.118.88', 'publicDnsName': 'ec2-52-20-118-88.compute-1.amazonaws.com', 'ipOwnerId': '735935705556'}, 'privateIpAddress': '172.31.59.85'}}, 'ebsOptimized': 'false', 'launchTime': '2016-07-13T02:32:00.000Z', 'architecture': 'x86_64', 'hypervisor': 'xen', 'rootDeviceType': 'ebs', 'ipAddress': '52.20.118.88', 'rootDeviceName': '/dev/sda1'}}}]
[ERROR   ] Public IP not detected.  If private IP is meant for bootstrap you must specify "ssh_interface: private_ips" in your profile.
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'_stamp': '2016-07-13T02:32:00.859942', 'event': 'setting tags', 'tags': {'Name': 'six.domain.com'}}
[DEBUG   ] Tags to set for six.domain.com: {'Name': 'six.domain.com'}
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=CreateTags&ResourceId.1=i-e3b2477c&Tag.0.Key=Name&Tag.0.Value=six.domain.com&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=CreateTags&ResourceId.1=i-e3b2477c&Tag.0.Key=Name&Tag.0.Value=six.domain.com&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=DescribeTags&Filter.1.Name=resource-id&Filter.1.Value=i-e3b2477c&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=DescribeTags&Filter.1.Name=resource-id&Filter.1.Value=i-e3b2477c&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[DEBUG   ] Setting the tags returned: [{'resourceType': 'instance', 'resourceId': 'i-e3b2477c', 'value': 'six.domain.com', 'key': 'Name'}]
[DEBUG   ] Using AWS endpoint: ec2.us-east-1.amazonaws.com
[DEBUG   ] AWS Request: https://ec2.us-east-1.amazonaws.com/?Action=ModifyNetworkInterfaceAttribute&Attachment.AttachmentId=eni-attach-209c3afd&Attachment.DeleteOnTermination=True&NetworkInterfaceId=eni-63c8e37d&Version=2014-10-01
[INFO    ] Starting new HTTPS connection (1): ec2.us-east-1.amazonaws.com
[DEBUG   ] Setting read timeout to None
[DEBUG   ] "GET /?Action=ModifyNetworkInterfaceAttribute&Attachment.AttachmentId=eni-attach-209c3afd&Attachment.DeleteOnTermination=True&NetworkInterfaceId=eni-63c8e37d&Version=2014-10-01 HTTP/1.1" 200 None
[DEBUG   ] AWS Response Status Code: 200
[INFO    ] Created node six.domain.com
[INFO    ] Salt node data. Private_ip: 172.31.59.85
[INFO    ] Salt interface set to: 172.31.59.85
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Sending event - data = {'_stamp': '2016-07-13T02:32:01.387591', 'ip_address': '172.31.59.85', 'event': 'waiting for ssh'}
[DEBUG   ] Attempting connection to host 172.31.59.85 on port 22
[DEBUG   ] Caught exception in wait_for_port: timed out
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 1)
[DEBUG   ] Caught exception in wait_for_port: timed out
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 2)
[DEBUG   ] Caught exception in wait_for_port: [Errno 111] Connection refused
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 3)
[DEBUG   ] Caught exception in wait_for_port: [Errno 111] Connection refused
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 4)
[DEBUG   ] Caught exception in wait_for_port: [Errno 111] Connection refused
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 5)
[DEBUG   ] Caught exception in wait_for_port: [Errno 111] Connection refused
[DEBUG   ] Retrying connection to host 172.31.59.85 on port 22 (try 6)
[DEBUG   ] Using /root/.ssh/KEY_FILE.pem as the key_filename
[DEBUG   ] Attempting to authenticate as ubuntu (try 1 of 15)
[DEBUG   ] SSH command: 'ssh -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none -oPasswordAuthentication=no -oChallengeResponseAuthentication=no -oPubkeyAuthentication=yes -oIdentitiesOnly=yes -oKbdInteractiveAuthentication=no -i /root/.ssh/KEY_FILE.pem -oConnectTimeout=60 -p 22 ubuntu@172.31.59.85 date'
[DEBUG   ] Child Forked! PID: 16389  STDOUT_FD: 4  STDERR_FD: 7
[DEBUG   ] Terminal Command: /bin/sh -c ssh -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oControlPath=none -oPasswordAuthentication=no -oChallengeResponseAuthentication=no -oPubkeyAuthentication=yes -oIdentitiesOnly=yes -oKbdInteractiveAuthentication=no -i /root/.ssh/KEY_FILE.pem -oConnectTimeout=60 -p 22 ubuntu@172.31.59.85 date
Warning: Permanently added '172.31.59.85' (ECDSA) to the list of known hosts.
[DEBUG   ] Warning: Permanently added '172.31.59.85' (ECDSA) to the list of known hosts.
Tue Jul 12 21:33:23 EST 2016
[DEBUG   ] Tue Jul 12 21:33:23 EST 2016
[INFO    ] Rendering deploy script: /usr/lib/python2.7/dist-packages/salt/cloud/deploy/bootstrap-salt.sh
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pull.ipc

...

six.domain.com:
    ----------
    amiLaunchIndex:
        0
    architecture:
        x86_64
    blockDeviceMapping:
        ----------
        item:
            ----------
            deviceName:
                /dev/sda1
            ebs:
                ----------
                attachTime:
                    2016-07-13T02:32:00.000Z
                deleteOnTermination:
                    true
                status:
                    attached
                volumeId:
                    vol-d7b3c004
    clientToken:
        None
    deployed:
        True
    dnsName:
        ec2-52-20-118-88.compute-1.amazonaws.com
    ebsOptimized:
        false
    groupSet:
        ----------
        item:
            ----------
            groupId:
                sg-XXXXXXXXX
            groupName:
                LEMP Stack
    hypervisor:
        xen
    id:
        i-e3b2477c
    image:
        ami-XXXXXXXX
    imageId:
        ami-XXXXXXXX
    instanceId:
        i-e3b2477c
    instanceState:
        ----------
        code:
            16
        name:
            running
    instanceType:
        t2.micro
    ipAddress:
        52.20.118.88
    keyName:
        KEY_NAME
    launchTime:
        2016-07-13T02:32:00.000Z
    monitoring:
        ----------
        state:
            disabled
    name:
        six.domain.com
    networkInterfaceSet:
        ----------
        item:
            ----------
            association:
                ----------
                ipOwnerId:
                    735935705556
                publicDnsName:
                    ec2-52-20-118-88.compute-1.amazonaws.com
                publicIp:
                    52.20.118.88
            attachment:
                ----------
                attachTime:
                    2016-07-13T02:32:00.000Z
                attachmentId:
                    eni-attach-209c3afd
                deleteOnTermination:
                    true
                deviceIndex:
                    0
                status:
                    attached
            description:
                None
            groupSet:
                ----------
                item:
                    ----------
                    groupId:
                        sg-XXXXXXXXX
                    groupName:
                        LEMP Stack
            macAddress:
                12:ce:24:6d:30:9b
            networkInterfaceId:
                eni-63c8e37d
            ownerId:
                735935705556
            privateDnsName:
                ip-172-31-59-85.ec2.internal
            privateIpAddress:
                172.31.59.85
            privateIpAddressesSet:
                ----------
                item:
                    ----------
                    association:
                        ----------
                        ipOwnerId:
                            735935705556
                        publicDnsName:
                            ec2-52-20-118-88.compute-1.amazonaws.com
                        publicIp:
                            52.20.118.88
                    primary:
                        true
                    privateDnsName:
                        ip-172-31-59-85.ec2.internal
                    privateIpAddress:
                        172.31.59.85
            sourceDestCheck:
                true
            status:
                in-use
            subnetId:
                subnet-XXXXXXXX
            vpcId:
                vpc-9bd87aff
    placement:
        ----------
        availabilityZone:
            us-east-1c
        groupName:
            None
        tenancy:
            default
    privateDnsName:
        ip-172-31-59-85.ec2.internal
    privateIpAddress:
        172.31.59.85
    private_ips:
        172.31.59.85
    productCodes:
        None
    public_ips:
        52.20.118.88
    reason:
        None
    rootDeviceName:
        /dev/sda1
    rootDeviceType:
        ebs
    size:
        t2.micro
    sourceDestCheck:
        true
    state:
        running
    subnetId:
        subnet-XXXXXXXX
    tagSet:
        ----------
        item:
            ----------
            key:
                Name
            value:
                six.domain.com
    virtualizationType:
        hvm
    vpcId:
        vpc-9bd87aff
```

**Comment by rallytime** [comment]
Ah, thanks for letting us know @lewayotte. I can reproduce this problem as well. I think we just need to gate that error message a little bit better.

**Comment by rallytime** [comment]
@lewayotte I have fixed this in #34641.

**Comment by lewayotte** [comment]
@rallytime I applied to fix to my server and it seems to have fixed it. Thanks!
